src/sfctl/entry.py

```diff
@@ -21,11 +23,21 @@ def cli():
                         commands_loader_cls=SFCommandLoader,
                         help_cls=SFCommandHelp)
 
+
 def launch():
     """Entry point for Service Fabric CLI.
 
     Configures and invokes CLI with arguments passed during the time the python
-    session is launched"""
+    session is launched.
+
+    This is run every time a sfctl command is invoked."""
 
     cli_env = cli()
-    return cli_env.invoke(sys.argv[1:])
+    invoke_return_value = cli_env.invoke(sys.argv[1:])
+
+    if invoke_return_value != 0 or ('cluster' and 'select' in sys.argv[1:]):
```

Comment:
Wrap this in a try except block. For example, we might not be able to read the file due to corruption. Do not fail call because of this. If file corruption occurs, do we handle clean up so operation can resume as usual?